src/main/java/com/opinta/service/ClientServiceImpl.java

```diff
@@ -112,17 +139,39 @@ public Client getEntityByUuidAnonymous(UUID uuid) throws IncorrectInputDataExcep
         }
         return client;
     }
-
+    
+    @Override
     @Transactional
     public Client saveEntity(Client client, User user) throws IncorrectInputDataException, AuthException {
+        client.setPostId(null);
         validateInnerReferencesAndFillObjectFromDB(client, user);
         client.setPhone(phoneService.getOrCreateEntityByPhoneNumber(client.getPhone().getPhoneNumber())
                 .removeNonNumericalCharacters());
         userService.authorizeForAction(client, user);
         log.info(saveLogEndpoint(Client.class, client));
         return clientDao.save(client);
     }
-
+    
+    @Override
+    @Transactional
+    public ClientDto updatePostId(UUID uuid, ClientType clientType, User user)
+            throws IncorrectInputDataException, AuthException {
+        Client client = getEntityByUuid(uuid, user);
+        if (client.getPostId() != null) {
+            return clientMapper.toDto(client);
+        }
+        client.setPostId(generatePostId(clientType));
+        clientDao.update(client);
+        return clientMapper.toDto(client);
+    }
+    
+    private String generatePostId(ClientType clientType) throws IncorrectInputDataException {
+        String yearString = valueOf(now().getYear());
+        yearString = yearString.substring(yearString.length() - 2);
+        return characterOf(clientType) + yearString +
+                postIdInnerNumberGenerator.generateNextNumber() + generateRandomChars(3, true);
```

Comment:
I can't check it, but I'm confident sure you can put this
    postIdInnerNumberGenerator.generateNextNumber()
in the line above